src/Objects/_PVector.js

```diff
@@ -0,0 +1,283 @@
+      } else if (len === 1) {
+        this.set(v.x || v[0] || 0,
+                 v.y || v[1] || 0,
+                 v.z || v[2] || 0);
+      } else { argsErr('set', len, 1); }
+      return this;
+    },
+    array: function() {
+      return [this.x, this.y, this.z];
+    },
+    normalize: function(target) {
+      const m = this.mag(),
+            canDivide = m !== 0 && m !== 1;
+      if (!arguments.length) { return canDivide? this.div(m) : this; }
+      return canDivide? _PVector.div(this, m, target)
+                      : target? target.set(this) : this.copy();
+    },
+    limit: function(max) {
+      return this.magSq() > max*max? this.normalize().mult(max) : this;
+    },
+    heading: function() {
+      //return -Math.atan2(-this.y, this.x);
+      return Math.atan2(this.y, this.x);
+    },
+    heading2D: null, // @Deprecated
+    mag: function() {
+      return Math.sqrt(this.magSq());
+    },
+    magSq: function() {
+      return sq(this.x) + sq(this.y) + sq(this.z);
+    },
+    setMag: function(target, length) {
+      const len = arguments.length;
+      return len === 1? this.normalize().mult(target) :
+             len > 1? this.normalize(target).mult(length) :
+             argsErr('setMag', len, 1);
+    },
+    rotate: function(angle) {
+      const prev_x = this.x,
+            c = Math.cos(angle),
+            s = Math.sin(angle);
+      this.x = c*this.x - s*this.y;
+      this.y = s*prev_x + c*this.y;
+      return this;
+    },
+    dist: function(v1, v2) {
+      return v2? _PVector.dist(v1, v2) : _PVector.dist(this, v1);
+    },
+    distSq: function(v1, v2) {
+      return v2? _PVector.distSq(v1, v2) : _PVector.distSq(this, v1);
+    },
+    dot: function(v, y, z) {
+      const len = arguments.length;
+      return len === 1? _PVector.dot(this, v) :
+             len === 2? _PVector.dot(v, y) :
+             len > 2? this.x*v + this.y*y + this.z*z :
+             argsErr('dot', len, 1);
+    },
+    cross: function(v1, v2, target) {
+      return target? _PVector.cross(v1, v2, target)
+                   : _PVector.cross(this, v1, v2);
+    },
+    lerp: function() {
+      var x, y, z, amt;
+      const len = arguments.length;
+      if ((len | 1) === 1) { argsErr('lerp', len, 2); }
+      if (len === 2) { // given vector and amt
+        const v = arguments[0];
+        x = v.x;
+        y = v.y;
+        z = v.z;
+        amt = arguments[1];
+      } else if (len === 3) { // given vector 1, vector 2 and amt
+        return _PVector.lerp(arguments[0], arguments[1], arguments[2]);
+      } else { // given x, y, z and amt
+        x = arguments[0];
+        y = arguments[1];
+        z = arguments[2];
+        amt = arguments[3];
+      }
+      this.x = lerp(this.x, x, amt);
+      this.y = lerp(this.y, y, amt);
+      this.z = lerp(this.z, z, amt);
+      return this;
+    },
+    add: function(v, y, z) {
+      if (y instanceof _PVector) { return _PVector.add(v, y, z); }
```

Comment:
Yes, so that should be fine then? If `y.add` exists, we know `y` is a PVector. It simply uses the JavaScript way to figure that out, rather than using the `instanceof` operator, which exists, but isn't really used in modern JS becuase JS can do feature detection much more efficiently.

Are there cases where `y.add` exists, but `y` is not a PVector?